browser/playlist/playlist_tab_helper.cc

```diff
@@ -123,6 +129,41 @@ void PlaylistTabHelper::OnItemCreated(mojom::PlaylistItemPtr item) {
   }
 }
 
+void PlaylistTabHelper::OnItemAddedToList(const std::string& playlist_id,
+                                          const std::string& item_id) {
+  auto iter = base::ranges::find_if(saved_items_, [&item_id](const auto& item) {
+    return item->id == item_id;
+  });
+
+  if (iter == saved_items_.end()) {
+    return;
+  }
+
+  (*iter)->parents.push_back(playlist_id);
+
+  for (auto& observer : observers_) {
+    observer.OnSavedItemsChanged(saved_items_);
+  }
+}
+
+void PlaylistTabHelper::OnItemRemovedFromList(const std::string& playlist_id,
```

Comment:
Q) don't we need to remove item from `saved_items_` when in this callback?

Reply:
As a item could belong to multiple playlists, we shouldn't erase the item. Instead, we'll do that when OnItemDeleted() is called. When there's no playlist references a item, this callback will be called.